webapp/channels/src/components/emoji/emoji_list_item/emoji_list_item.tsx

```diff
@@ -22,72 +22,69 @@ export type Props = {
     actions: {
         deleteCustomEmoji: (emojiId: string) => void;
     };
-}
+};
 
-export default class EmojiListItem extends React.PureComponent<Props> {
-    static defaultProps = {
-        emoji: {} as CustomEmoji,
-        currentUserId: '',
-        creatorDisplayName: '',
-    };
-
-    handleDelete = (): void => {
-        if (!this.props.emoji) {
+const EmojiListItem = ({
+    actions: {
+        deleteCustomEmoji,
+    },
+    onDelete,
+    emoji = {} as CustomEmoji,
+    creatorUsername,
+    currentUserId = '',
+    creatorDisplayName = '',
+}: Props) => {
+    const handleDelete = (): void => {
+        if (!emoji) {
             return;
         }
-
-        if (this.props.onDelete) {
-            this.props.onDelete(this.props.emoji.id);
+        if (onDelete) {
+            onDelete(emoji.id);
         }
-
-        this.props.actions.deleteCustomEmoji(this.props.emoji.id);
+        deleteCustomEmoji(emoji.id);
     };
 
-    render(): JSX.Element {
-        const emoji = this.props.emoji as CustomEmoji;
-        const creatorUsername = this.props.creatorUsername;
-        let creatorDisplayName = this.props.creatorDisplayName;
-
-        if (creatorUsername && creatorUsername !== creatorDisplayName) {
-            creatorDisplayName += ' (@' + creatorUsername + ')';
-        }
+    if (creatorUsername && creatorUsername !== creatorDisplayName) {
+        // eslint-disable-next-line no-param-reassign
+        creatorDisplayName += ' (@' + creatorUsername + ')';
+    }
 
-        let deleteButton = <DeleteEmojiButton onDelete={this.handleDelete}/>;
+    let deleteButton = <DeleteEmojiButton onDelete={handleDelete}/>;
 
-        if (emoji.creator_id === this.props.currentUserId) {
-            deleteButton = (
-                <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
+    if (emoji.creator_id === currentUserId) {
+        deleteButton = (
+            <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
```

Comment:
To avoid passing a new list on every render, put these as constants outside of the component:

```
const DELETE_PERMISSION = [Permissions.DELETE_EMOJIS];
const DELETE_OTHER_PERMISSION = [Permissions.DELETE_OTHER_EMOJIS];

const EmojiListItem = ({
   ...
```